LoRaEngine/test/LoRaWan.IntegrationTest/IntegrationTestFixture.cs

```diff
@@ -296,9 +324,16 @@ internal async Task<Twin> GetTwinAsync(string deviceId)
             return await GetRegistryManager().GetTwinAsync(deviceId);            
         }
 
-        internal async Task SendCloudToDeviceMessage(string deviceId, string messageText)
+        internal async Task SendCloudToDeviceMessage(string deviceId, string messageText, Dictionary<String,String> messageProperties)
```

Comment:
Tip 1: when adding an optional parameter to an existing method you can set a default value. In that case you don't have to fix all the other callers that will pass the default parameter. 

In this example:
```
internal async Task SendCloudToDeviceMessage(string deviceId, string messageText, Dictionary<String,String> messageProperties = null)
```

Tip 2: When you don't need an concrete collection implementation it is a good practice to use interfaces:
Dictionary<> &rarr; IDictionary<>
List<> &rarr; ICollection<> or IEnumerable<> (this way you can use with arrays, lists, hashsets, etc)